src/SBP_coefficients/WilliamsDuru2024.jl

```diff
@@ -0,0 +1,766 @@
+"""
+    WilliamsDuru2024(version::Symbol)
+
+Coefficients of the upwind SBP operators given in
+- Williams, Duru (2024)
+  Full-Spectrum Dispersion Relation Preserving Summation-By-Parts Operators.
+  SIAM Journal on Numerical Analysis 62.4, pp. 1565-1588.
+
+You can choose between the different versions `:central`, `:plus`, and `:minus`.
+"""
+struct WilliamsDuru2024 <: SourceOfCoefficients
+    kind::Symbol
+
+    function WilliamsDuru2024(kind::Symbol)
+        if (kind !== :plus) && (kind !== :minus) && (kind !== :central)
+            throw(ArgumentError("The only choices are :plus, :minus, and :central, not :$kind."))
+        end
+        new(kind)
+    end
+end
+
+function Base.show(io::IO, source::WilliamsDuru2024)
+    if get(io, :compact, false)
+        summary(io, source)
+    else
+        print(io,
+            "Williams, Duru (2024) \n",
+            "  Full-Spectrum Dispersion Relation Preserving Summation-By-Parts Operators. \n",
+            "  SIAM Journal on Numerical Analysis 62.4, pp. 1565-1588. \n",
+            "  (upwind coefficients ", source.kind, ")")
+    end
+end
+
+
+function first_derivative_coefficients(source::WilliamsDuru2024, order::Int,
+                                       T=Float64, mode=FastMode())
+    if order == 4
+        left_boundary_plus = (
+            DerivativeCoefficientRow{T,1, 6}(SVector( T(-7244216288712856142349841908536610983621117531423//4383417301651166629768772548289373010602184474830),
```

Comment:
I guess these coefficients are rational approximations of the floating point values from the paper. Do they still satisfy the accuracy conditions exactly?

Reply:
I got the rational coefficients directly from Kenneth Duru. I added some rational tests that show they satisfy the dual-paring SBP property and accuracy conditions exactly.

Reply:
Great, thanks 👍